core/html_api.php

```diff
@@ -369,27 +369,98 @@ function html_top_banner() {
 }
 
 /**
- * A function that outputs that an operation was successful and provides a redirect link.
- * @param string $p_redirect_url The url to redirect to.
- * @param string $p_message      Message to display to the user.
+ * Outputs an operation successful message with multiple redirection buttons.
+ * @param array  $p_buttons      Array of (URL, label) pairs used to generate
+ *                               the buttons; if label is null or unspecified,
+ *                               the default 'proceed' text will be displayed.
+ * @param string $p_message      Message to display to the user. If none is
+ *                               provided, a default message will be printed
+ * @param string $p_type         One of the constants CONFIRMATION_TYPE_SUCCESS,
+ *                               CONFIRMATION_TYPE_WARNING, CONFIRMATION_TYPE_FAILURE
  * @return void
  */
-function html_operation_successful( $p_redirect_url, $p_message = '' ) {
+function html_operation_successful_buttons( array $p_buttons, $p_message = '', $p_type = CONFIRMATION_TYPE_SUCCESS ) {
+	switch( $p_type ) {
+		case CONFIRMATION_TYPE_FAILURE:
+			$t_alert_css = 'alert-danger';
+			$t_message = 'operation_failed';
+			break;
+		case CONFIRMATION_TYPE_WARNING:
+			$t_message = 'operation_warnings';
```

Comment:
For better readability and to be consistent with other cases.
Insert this line after `$t_alert_css = 'alert-warning';`